staging/vhost-device-sound/src/audio_backends/pipewire.rs

```diff
@@ -331,54 +344,106 @@ impl AudioBackend for PwBackend {
                 .process(move |stream, _data| match stream.dequeue_buffer() {
                     None => debug!("No buffer recieved"),
                     Some(mut buf) => {
-                        let datas = buf.datas_mut();
-                        let frame_size = info.channels * size_of::<i16>() as u32;
-                        let data = &mut datas[0];
-                        let n_bytes = if let Some(slice) = data.data() {
-                            let mut n_bytes = slice.len();
-                            let mut streams = streams.write().unwrap();
-                            let streams = streams
-                                .get_mut(stream_id as usize)
-                                .expect("Stream does not exist");
-                            let Some(buffer) = streams.buffers.front_mut() else {
-                                return;
-                            };
-
-                            let mut start = buffer.pos;
-
-                            let avail = (buffer.desc_len() - start as u32) as i32;
-
-                            if avail < n_bytes as i32 {
-                                n_bytes = avail.try_into().unwrap();
+                        match spa_dir {
+                            spa::Direction::Input => {
+                                let datas = buf.datas_mut();
+                                let data = &mut datas[0];
+                                let mut n_samples = data.chunk().size() as usize;
+                                if let Some(slice) = data.data() {
```

Comment:
```suggestion
                                let Some(slice) = data.data() else {
                                    return 0;
                                }
```

Reply:
To avoid the extra nesting

Reply:
Ah, looks like need just `return;`? Then the `else { 0 };` later is unnecessary.